emails/tests/views_tests.py

```diff
@@ -307,36 +306,34 @@ def setUp(self) -> None:
         send_raw_email_patcher.start()
         self.addCleanup(send_raw_email_patcher.stop)
 
-    def get_details_from_mock_send_raw_email(
+    def assert_email_equals(
         self,
-    ) -> tuple[str, str, dict[str, str], str]:
+        name: str,
+        sender: str | None = None,
+        recipient: str | None = None,
+        replace_mime_boundaries: bool = False,
+    ) -> None:
         """
-        Get sender, recipient, headers, and email from the message sent by mocked
-        ses_client.send_raw_email
+        Extract the email and check against the expected email.
+
+        name: the name of the test fixture to check against the email
+        sender: if set, assert that the sender email matches this string
+        recipient: if set, assert that the recipient email matches this string
+        replace_mime_boundaries: if true, replace randomized MIME boundaries with
+        sequential test versions
         """
         self.mock_send_raw_email.assert_called_once()
         source = self.mock_send_raw_email.call_args[1]["Source"]
         destinations = self.mock_send_raw_email.call_args[1]["Destinations"]
         assert len(destinations) == 1
         raw_message = self.mock_send_raw_email.call_args[1]["RawMessage"]["Data"]
-        headers: dict[str, str] = {}
-        last_key = None
-        for line in raw_message.splitlines():
-            if not line:
-                # Start of message body, done with headers
-                return source, destinations[0], headers, raw_message
-            if line[0] in (" ", "\t"):
-                # Continuation of last header
-                assert last_key
-                headers[last_key] += line
-            else:
-                # New headers
-                assert ": " in line
-                key, val = line.split(": ", 1)
-                assert key not in headers
-                headers[key] = val
-                last_key = key
-        raise Exception("Never found message body!")
+        if "" not in raw_message.splitlines():
+            raise Exception("Never found message body!")
+        if sender is not None:
+            assert source == sender
+        if recipient is not None:
+            assert destinations[0] == recipient
+        assert_email_equals(raw_message, name, replace_mime_boundaries)
```

Comment:
suggestion (non-blocking): Maybe it's just because I'm looking at the changeset and not the whole file, but this function name first made me think this was a recursive call, because there's a module-level function with the same name as this class method. Maybe change this to be more explicit?

Reply:
Thanks, I had the same itch. The new functions:

```python
def assert_email_equals_fixture(
    output_email: str, fixture_name: str, replace_mime_boundaries: bool = False
) -> None:
    """Focuses on the single assertion."""
    ...

class SNSNotificationTestBase(TestCase):
    def check_sent_email_matches_fixture(
        self,
        fixture_name: str,
        replace_mime_boundaries: bool = False,
        expected_source: str | None = None,
        expected_destination: str | None = None,
    ) -> None:
    """Checks the call to send_raw_email, and call assert_email_equals_fixture."""
    ...
```